packages/server/src/koaJsonRpc/lib/httpErrorMapper.ts

```diff
@@ -54,7 +52,7 @@ export function translateRpcErrorToHttpStatus(jsonRpcError: JsonRpcError | JsonR
   // Handle Mirror Node errors (-32020)
   // Note: -32020 corresponds to predefined.MIRROR_NODE_UPSTREAM_FAILURE,
   // where `jsonRpcError.data` represents the actual HTTP status code returned from the Mirror Node upstream server.
-  if (jsonRpcError.code === -32020 && jsonRpcError.data) {
+  if (jsonRpcError.code === -32020 && typeof jsonRpcError.data === 'number') {
```

Comment:
q: What else besides HTTP status code can be the `.data` field? Can we rename it to `.statusCode`?

Reply:
At the moment, nothing. However, the idea is that this `.data` can be a generic field to include any additional information.

More practically though, the issue is that `JsonRpcError` class in the Relay needs to match the type `IJsonRpcError` interface in server. Otherwise the following `jsonRespError` will fail

https://github.com/hiero-ledger/hiero-json-rpc-relay/blob/539ed01672908706422cd83919957eb14fabe785/packages/server/src/koaJsonRpc/index.ts#L181-L187

If we rename it, we need to do some conversion just for this case.